src/components/tx/ApprovalEditor/ApprovalValueField.tsx

```diff
@@ -0,0 +1,109 @@
+import NumberField from '@/components/common/NumberField'
+import TokenIcon from '@/components/common/TokenIcon'
+import { shortenAddress } from '@/utils/formatters'
+import { validateAmount, validateDecimalLength } from '@/utils/validation'
+import { Autocomplete, Box, type MenuItemProps, Typography, MenuItem } from '@mui/material'
+import { useController, useFormContext } from 'react-hook-form'
+import css from './styles.module.css'
+import { PSEUDO_APPROVAL_VALUES, type ApprovalInfo } from './utils/approvals'
+
+const ApprovalOption = ({ menuItemProps, value }: { menuItemProps: MenuItemProps; value: string }) => {
+  return (
+    <MenuItem key={value} {...menuItemProps}>
+      {value}
+    </MenuItem>
+  )
+}
+
+export const ApprovalValueField = ({
+  name,
+  readonly = false,
+  tx,
+}: {
+  name: string
+  readonly?: boolean
+  tx: ApprovalInfo
+}) => {
+  const { control } = useFormContext() || {}
```

Comment:
Let's remove the fallback so that it errors in development if not embedded within a form.

```suggestion
  const { control } = useFormContext()
```